internal/repository/instance/domain.go

```diff
@@ -43,6 +47,29 @@ func (e *DomainAddedEvent) UniqueConstraints() []*eventstore.UniqueConstraint {
 	return []*eventstore.UniqueConstraint{NewAddInstanceDomainUniqueConstraint(e.Domain)}
 }
 
+func (e *DomainAddedEvent) Fields() []*eventstore.FieldOperation {
+	return []*eventstore.FieldOperation{
+		eventstore.SetField(
+			e.Aggregate(),
+			domainSearchObject(e.Domain),
+			InstanceDomainSearchField,
+			&eventstore.Value{
+				Value:        e.Domain,
+				MustBeUnique: true,
```

Comment:
Off course, this is already handled by the eventstore. So it might not be a problem. Just think it over and feel free to dismiss my comment if you do not believe it can result in problems:

https://github.com/zitadel/zitadel/blob/77923b4ec657fe756ab2601f0ae53126872acab6/internal/repository/instance/domain.go#L46-L48

Reply:
you are absolutely right. I think we should have a look at that if we decide to go with the fields table. I created an issue: https://github.com/zitadel/zitadel/issues/9009